ci/Dockerfile

```diff
@@ -1,46 +1,131 @@
+
+# Download & install linuxdeployqt
+RUN /tmp/build/scripts/install-linuxdeployqt.sh
+
+# Reconfigure locale
+RUN locale-gen en_US.UTF-8 && dpkg-reconfigure locales \
+# Add group & user
+    && groupadd -r user && useradd --create-home --gid user user && echo 'user ALL=NOPASSWD: ALL' > /etc/sudoers.d/user
+
+# Copy QT toolchain
+RUN chmod -R 777 /opt && mkdir -p /opt/qt/$QT_VERSION
+COPY --from=qt-install /home/user/$QT_VERSION /opt/qt/$QT_VERSION/.
+
+USER user
+WORKDIR /home/user
+ENV HOME /home/user
 
-# $QT_PATH and $QT_PLATFORM are provided by the docker image
 # $QT_PATH/$QT_VERSION/$QT_PLATFORM/bin is already prepended to $PATH
-# However $QT_VERSION is not exposed to environment so set it here
-ENV QT_VERSION="5.14.0"
 ENV QTDIR="${QT_PATH}/${QT_VERSION}/${QT_PLATFORM}"
 ENV LD_LIBRARY_PATH="${QTDIR}/lib:${LD_LIBRARY_PATH}"
-# $OPENSSL_PREFIX is provided by the docker image
-ENV LIBRARY_PATH="${OPENSSL_PREFIX}/lib:${LIBRARY_PATH}"
-ENV OPENSSL_ROOT_DIR="${OPENSSL_PREFIX}"
-
-RUN export DEBIAN_FRONTEND=noninteractive \
- && sudo apt update -yq \
- && sudo apt install -yq software-properties-common \
- && sudo add-apt-repository -y ppa:git-core/ppa \
- && sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test \
- && sudo apt update -yq \
- && sudo apt purge -yq gnupg \
- && sudo apt install -yq --fix-missing \
-      jq git s3cmd gnupg2 ca-certificates \
-      build-essential extra-cmake-modules gcc-9 g++-9 \
-      libpcre3-dev libnss3 libxcomposite1 libxtst6 libpcsclite-dev \
-      gstreamer1.0-plugins-good gstreamer1.0-plugins-bad \
-      gstreamer1.0-plugins-ugly gstreamer1.0-libav \
-      gstreamer1.0-tools gstreamer1.0-alsa gstreamer1.0-pulseaudio
-
-# C++17 is required
-RUN sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 100 \
- --slave /usr/bin/g++ g++ /usr/bin/g++-9
-
-# CMake 3.16 or higher is required
-RUN wget https://github.com/Kitware/CMake/releases/download/v3.22.1/cmake-3.22.1-linux-x86_64.sh \
- && chmod +x cmake-3.22.1-linux-x86_64.sh \
- && sudo mkdir /opt/cmake \
- && sudo sh cmake-3.22.1-linux-x86_64.sh --prefix=/opt/cmake --exclude-subdir --skip-license \
- && rm -f cmake-3.22.1-linux-x86_64.sh 
-ENV PATH="/opt/cmake/bin:${PATH}"
 
 # Fix LetsEncrypt X3 root certificate expliration
 RUN sudo sed -i 's#\(mozilla/DST_Root_CA_X3.crt\)#!\1#' /etc/ca-certificates.conf \
  && sudo dpkg-reconfigure -fnoninteractive ca-certificates
 
+# CMake 3.16 or higher is required
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.22.1/cmake-3.22.1-linux-x86_64.sh \
+    && chmod +x cmake-3.22.1-linux-x86_64.sh \
+    && sudo mkdir /opt/cmake \
+    && ls -l \
+    && sudo sh cmake-3.22.1-linux-x86_64.sh --prefix=/opt/cmake --exclude-subdir --skip-license \
+    && rm -f cmake-3.22.1-linux-x86_64.sh 
+ENV PATH="/opt/cmake/bin:${PATH}"
+
 # Installing Golang
 RUN GOLANG_SHA256="9e5de37f9c49942c601b191ac5fba404b868bfc21d446d6960acc12283d6e5f2" \
  && GOLANG_TARBALL="go1.18.5.linux-amd64.tar.gz" \
```

Comment:
This could get the latest too

Reply:
Sure, we can do it if Go experts agree on it - @richard-ramos

Reply:
go1.19.5 should be fine

Reply:
done